diffpy/snmf/subroutines.py

```diff
@@ -242,52 +242,69 @@ def reconstruct_data(stretching_factor_matrix, component_matrix, weight_matrix,
     component_matrix = np.asarray(component_matrix)
     weight_matrix = np.asarray(weight_matrix)
     stretched_component_series = []
-    for moment in range(moment_amount):
-        for component in range(component_amount):
+    for moment in range(number_of_moments):
+        for component in range(number_of_components):
             stretched_component = get_stretched_component(stretching_factor_matrix[component, moment],
                                                           component_matrix[:, component], signal_length)
             stretched_component_series.append(stretched_component)
     stretched_component_series = np.column_stack(stretched_component_series)
 
     reconstructed_data = []
     moment = 0
-    for s_component in range(0, moment_amount * component_amount, component_amount):
-        block = stretched_component_series[:, s_component:s_component + component_amount]
-        for component in range(component_amount):
+    for s_component in range(0, number_of_moments * number_of_components, number_of_components):
+        block = stretched_component_series[:, s_component:s_component + number_of_components]
+        for component in range(number_of_components):
             block[:, component] = block[:, component] * weight_matrix[component, moment]
         reconstructed_data.append(block)
         moment += 1
     return np.column_stack(reconstructed_data)
 
 
-def update_stretching_matrix(stretching_factor_matrix, weight_matrix, component_matrix, data_input, moment_amount,
-                             component_amount, signal_length, smoothness, sparsity, smoothness_term):
-    """
+def update_stretching_matrix(stretching_factor_matrix, weight_matrix, component_matrix, data_input, number_of_moments,
+                             number_of_components, signal_length, smoothness, sparsity, smoothness_term):
+    """Updates the stretching factor matrix
+
+    Calculates a new stretching factor matrix by finding the argument (stretching factor matrix) that minimizes
+    objective function.
 
     Parameters
     ----------
-    stretching_factor_matrix
-    weight_matrix
-    component_matrix
-    data_input
-    moment_amount
-    component_amount
-    signal_length
-    smoothness
-    sparsity
-    smoothness_term
+    stretching_factor_matrix: 2d array like
+      The current stretching factor matrix.
+
+    weight_matrix: 2d array like
+      The matrix containing the weightings of the component signals.
+
+    component_matrix: 2d array like
+      The matrix containing the calculated component signals.
+
```

Comment:
is having empty lines here part of the numpy standard?  For me this stretches things out too much but I would rather follow the standard